desc/integrals/bounce_integral.py

```diff
@@ -1,26 +1,818 @@
+        Preferably power of 2 for ``grid.num_theta`` and ``grid.num_zeta``.
+    nodes : jnp.ndarray
+        Shape (L, M, N, 2) or (M, N, 2).
+        DESC coordinates (θ, ζ) sourced from the Clebsch coordinates
+        ``FourierChebyshevSeries.nodes(M,N,domain=(0,2*jnp.pi))``.
+    f : jnp.ndarray
+        Function evaluated on ``grid``.
+
+    Returns
+    -------
+    a : FourierChebyshevSeries
+        Spectral coefficients of f(α, ζ).
+
+    """
+    assert nodes.shape[-1] == 2
+    if not is_reshaped:
+        f = grid.meshgrid_reshape(f, "rtz")
+
+    M, N = nodes.shape[-3], nodes.shape[-2]
+    return FourierChebyshevSeries(
+        f=interp_rfft2(
+            # Interpolate to nodes in Clebsch space,
+            # which is not a tensor product node set in DESC space.
+            xq=nodes.reshape(*nodes.shape[:-3], M * N, 2),
+            f=f[..., jnp.newaxis, :, :],
+            axes=(-1, -2),
+        ).reshape(*nodes.shape[:-3], M, N),
+        domain=(0, 2 * jnp.pi),
+    )
+
+
+def _transform_to_clebsch_1d(grid, alpha, theta, B, N_B, is_reshaped=False):
+    """Transform to single variable Clebsch spectral domain.
+
+    Parameters
+    ----------
+    grid : Grid
+        Tensor-product grid in (θ, ζ) with uniformly spaced nodes in
+        (2π × 2π) poloidal and toroidal coordinates.
+        Preferably power of 2 for ``grid.num_theta`` and ``grid.num_zeta``.
+    alpha : jnp.ndarray
+        Shape (L, num_transit) or (num_transit, ).
+        Sequence of poloidal coordinates A = (α₀, α₁, …, αₘ₋₁) that specify field line.
+    theta : jnp.ndarray
+        Shape (L, M, N) or (M, N).
+        DESC coordinates θ sourced from the Clebsch coordinates
+        ``FourierChebyshevSeries.nodes(M,N,domain=(0,2*jnp.pi))``.
+    B : jnp.ndarray
+        |B| evaluated on ``grid``.
+    N_B : int
+        Desired Chebyshev spectral resolution for |B|. Preferably power of 2.
+
+    Returns
+    -------
+    T, B : tuple[PiecewiseChebyshevSeries]
+        Set of 1D Chebyshev spectral coefficients of θ along field line.
+        {θ_α : ζ ↦ θ(α, ζ) | α ∈ A} where A = (α₀, α₁, …, αₘ₋₁) is ``alpha``.
+        Likewise with |B|.
+
+    """
+    if not is_reshaped:
+        B = grid.meshgrid_reshape(B, "rtz")
+
+    # Evaluating set of single variable maps is more efficient than evaluating
+    # multivariable map, so we project θ to a set of Chebyshev series.
+    T = FourierChebyshevSeries(f=theta, domain=(0, 2 * jnp.pi)).compute_cheb(alpha)
+    theta = T.evaluate(N_B)
+    xq = jnp.stack(
+        [theta, jnp.broadcast_to(cheb_pts(N_B, domain=T.domain), theta.shape)], axis=-1
+    ).reshape(*alpha.shape[:-1], alpha.shape[-1] * N_B, 2)
+    B = interp_rfft2(
+        xq=xq,
+        f=B[..., jnp.newaxis, :, :],
+        axes=(-1, -2),
+    ).reshape(*alpha.shape, N_B)
+    # Need |B| parameterized by single variable to compute roots.
+    B = PiecewiseChebyshevSeries(cheb_from_dct(dct(B, type=2, axis=-1)) / N_B, T.domain)
```

Comment:
todo: short circuit convergence of poloidal fourier series. evaluate each series at the connecting knot 2pi. Ensure perfect continuity by shifting the following chebyshev series up or down by the difference between the two chebyshev series that share the same knot. can just add to constant coefficient term of the following